features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/dbscripts/stored-procedures/oracle/sessiondata-cleanup/oracle-sessiondata-cleanup.sql

```diff
@@ -200,13 +159,155 @@ BEGIN
+            COMMIT;
+
+            EXIT WHEN rowcount <= checkcount;
+
+            IF ( tracingenabled ) THEN
+            EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),'' '')';
+            EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''TEMPORARY IDN_AUTH_USER_SESSION_TMP CHUNK TABLE CREATED WITH '||rowcount||''')';
+            COMMIT;
+            END IF;
+
+            LOOP
+                SELECT COUNT(1) INTO rowcount FROM all_tables WHERE owner = current_schema AND table_name = 'TEMP_USER_SESSION_BATCH';
+
+                IF ( rowcount = 1 ) THEN
+                    EXECUTE IMMEDIATE 'DROP TABLE TEMP_USER_SESSION_BATCH';
+                    COMMIT;
+                END IF;
+
+                EXECUTE IMMEDIATE 'CREATE TABLE TEMP_USER_SESSION_BATCH (SESSION_ID varchar(100),CONSTRAINT BATCH_IDN_USER_SESS_PRI PRIMARY KEY (SESSION_ID)) NOLOGGING';
+                COMMIT;
+
+                EXECUTE IMMEDIATE 'INSERT INTO TEMP_USER_SESSION_BATCH SELECT SESSION_ID FROM IDN_AUTH_USER_SESSION_TMP WHERE rownum <= :batchSize'
+                USING batchsize;
+                rowcount := SQL%rowcount;
+                COMMIT;
+
+                EXIT WHEN rowcount = 0;
+
+                IF ( tracingenabled ) THEN
+                EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''TEMPORARY TEMP_USER_SESSION_BATCH TABLE CREATED WITH '|| rowcount||''')';
+                COMMIT;
+                END IF;
+
+                -- Deleting user-session mappings from 'IDN_AUTH_USER_SESSION_MAPPING' table
+                EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_USER_SESSION_MAPPING WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
+                sessionmappingcleanupcount := SQL%rowcount;
+                COMMIT;
+                IF ( tracingenabled ) THEN
+                    deletedmappingsessions := deletedmappingsessions + sessionmappingcleanupcount;
+                    EXECUTE IMMEDIATE 'INSERT INTO LOG_WSO2_SESSION_CLEANUP_SP (TIMESTAMP,LOG) VALUES (TO_CHAR( SYSTIMESTAMP, ''DD.MM.YYYY HH24:MI:SS:FF4''),''DELETED USER-SESSION MAPPINGS COMPLETED WITH '|| sessionmappingcleanupcount||''')';
+                    COMMIT;
+                END IF;
+                -- End of deleting user-session mappings from 'IDN_AUTH_USER_SESSION_MAPPING' table
+
+                -- Deleting session app info from 'IDN_AUTH_SESSION_APP_INFO' table
+                SELECT COUNT(*) INTO ROWCOUNT from ALL_TABLES where OWNER = CURRENT_SCHEMA AND table_name = upper('IDN_AUTH_SESSION_APP_INFO');
+                IF (ROWCOUNT = 1) then
+                    EXECUTE IMMEDIATE 'DELETE FROM IDN_AUTH_SESSION_APP_INFO WHERE SESSION_ID IN (SELECT SESSION_ID FROM TEMP_USER_SESSION_BATCH)';
```

Comment:
use inner join

Reply:
Resolving due to the same reason as https://github.com/wso2/carbon-identity-framework/pull/3313#discussion_r556321872